modules/claims_api/app/controllers/claims_api/v1/forms/disability_compensation_controller.rb

```diff
@@ -350,22 +350,123 @@ def validate_form_526_disabilities!
           validate_form_526_disability_classification_code!
           validate_form_526_disability_approximate_begin_date!
           validate_form_526_special_issues!
+          validate_form_526_disability_secondary_disabilities!
+        end
+
+        def validate_form_526_disability_secondary_disability_disability_action_type!(disability)
+          return unless disability['disabilityActionType'] == 'NONE' && disability['secondaryDisabilities'].blank?
+
+          raise ::Common::Exceptions::InvalidFieldValue.new('disabilities.secondaryDisabilities',
+                                                            disability['secondaryDisabilities'])
+        end
+
+        def validate_form_526_disability_secondary_disability_classification_code!(secondary_disability)
+          return unless bgs_classification_ids.exclude?(secondary_disability['classificationCode'])
+
+          raise ::Common::Exceptions::InvalidFieldValue.new(
+            'disabilities.secondaryDisabilities.classificationCode',
+            secondary_disability['classificationCode']
+          )
+        end
+
+        def validate_form_526_disability_secondary_disability_classification_code_matches_name!(secondary_disability)
+          return unless secondary_disability['classificationCode'] != secondary_disability['name']
+
+          raise ::Common::Exceptions::InvalidFieldValue.new(
+            'disabilities.secondaryDisabilities.name',
+            secondary_disability['name']
+          )
+        end
+
+        def validate_form_526_disability_secondary_disability_name!(secondary_disability)
+          return if %r{([a-zA-Z0-9\-'.,/\(\)]([a-zA-Z0-9\-',. ])?)+$}.match?(secondary_disability['name']) &&
+                    secondary_disability['name'].length <= 255
+
+          raise ::Common::Exceptions::InvalidFieldValue.new(
+            'disabilities.secondaryDisabilities.name',
+            secondary_disability['name']
+          )
+        end
+
+        # rubocop:disable Metrics/MethodLength
+        def validate_form_526_disability_secondary_disability_approximate_begin_date!(secondary_disability)
```

Comment:
We talked about it offline, but FWIW, in a couple places, I took a different approach with these "approximateDate" fields that EVSS accepts.  Rather than our API accepting the optional components of an approximate date, I've been having our API accept valid date patterns. (ie. `yyyy-mm-dd`). Then I convert the valid date into the components required by EVSS.  That approach seemed cleaner to me, rather than the approach EVSS is taking where they accept components of a date and sometimes some of those date components are required based on the existence of other date components.  (i.e. You can send in just a `year`.  Or you can send in a `year` and a `month`.  Or you can send in a `year`, `month`, and `day`.  But you can't pass in a `day` and a `year`.).  That just felt like clunky and confusing to me.  🤷

Reply:
https://github.com/department-of-veterans-affairs/vets-api/pull/7805/commits/f4da4380e69443b9ae807c5d4c823a9ee4ca33c0